src/components/generate/index.js

```diff
@@ -0,0 +1,116 @@
+import { useState, useCallback, Fragment, useEffect } from 'react';
+import Input from '../input';
+import Select from '../select';
+import Button from '../button';
+import SnackBar from '../snackbar';
+import Loader from '../loader';
+import Error from '../error';
+import classes from './styles.module.css';
+import { isEmptyString, copyText } from '../../utils';
+import { SEVERITY } from '../constants';
+
+const Generate = () => {
+    const [inputTextVal, setInputTextVal] = useState('');
+    const [inputSecretKey, setInputSecretKey] = useState('');
+    const [validity, setValidity] = useState('');
+    const [loading, setLoading] = useState(false);
+    const [openSnackbar, setOpenSnackbar] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [errorStatus, setErrorStatus] = useState('');
+    const [snackbarMessage, setSnackbarMessage] = useState('');
+    const [snackbarSeverity, setSnackbarSeverity] = useState(SEVERITY.INFO);
+    const [fetchResult, setFetchResult] = useState(false);
+    const [url, setUrl] = useState('');
+    const [urlCopied, setUrlCopied] = useState(false);
+
+    useEffect(() => fetchResult && sendNewMessageRequest(), [fetchResult]);
+
+    const handleInputTextChange = useCallback(value => setInputTextVal(value), []);
+    const handleInputSecretKeyChange = useCallback(value => setInputSecretKey(value), []);
+    const handleValidityChange = useCallback(value => setValidity(value), []);
+    const handleSnackbarClose = useCallback(() => setOpenSnackbar(false), []);
+    const handleRedirectToHome = useCallback(() => {
+        setErrorMessage('');
+        setErrorStatus('');
+        setUrl('');
+    }, []);
+
+    const openInNewTab = useCallback(() => {
+        const newWindow = window.open(url, '_blank', 'noopener,noreferrer');
+        if (newWindow) newWindow.opener = null;
+    }, [url]);
+
+    const handleEncryption = useCallback(() => {
+        const message = (isEmptyString(inputTextVal) || isEmptyString(inputSecretKey) || isEmptyString(validity)) &&
+            'Missing either Text to Encrypt or Secret Key or Validity Duration!';
+        const severity = (isEmptyString(inputTextVal) || isEmptyString(inputSecretKey)) && SEVERITY.INFO;
+        setSnackbarMessage(message);
+        setSnackbarSeverity(severity);
+        setOpenSnackbar(!isEmptyString(message));
+        setFetchResult(isEmptyString(message) ? true : false);
+    }, [inputTextVal, inputSecretKey, validity]);
+
+    const sendNewMessageRequest = useCallback(() => {
+        setLoading(true);
+        handleInputTextChange('');
+        handleInputSecretKeyChange('');
+        fetch('http://localhost:5050/api/v1/messages', {
```

Comment:
use the url variable declared in constants file

Reply:
done